apps/api-consumer-producer/src/app/project/project.service.ts

```diff
@@ -0,0 +1,289 @@
+                                key_map["tagline"] = i
+                                break
+                        }
+                    })
+                } else {
+                    key_map["name"] = 0
+                    key_map["description"] = 1
+                    key_map["developer"] = 2
+                    key_map["status"] = 3
+                    key_map["category"] = 4
+                    key_map["github"] = 5
+                    key_map["homepage"] = 5
+                    key_map["link"] = 6
+                    key_map["commits"] = 7
+                    key_map["stars"] = 8
+                    key_map["network"] = 9
+                    key_map["tagline"] = 10
+                }
+
+                if (rows.length) {
+                    resolve(
+                        rows.map(row => {
+                            return {
+                                name: row[key_map["name"]] || "",
+                                description: row[key_map["description"]] || "",
+                                developer: row[key_map["developer"]] || "",
+                                status: ProjectStatus[row[key_map["status"]]] || 0,
+                                category: row[key_map["category"]] || "none",
+                                link: row[key_map["link"]] || "",
+                                commits: row[key_map["commits"]] || 0,
+                                stars: row[key_map["stars"]] || 0,
+                                externalLinks: [
+                                    { name: "github", url: row[key_map["github"]] || "" },
+                                    { name: "homepage", url: row[key_map["homepage"]] || "" },
+                                ],
+                                network: row[key_map["network"]] || "",
+                                tagline: row[key_map["tagline"]] || "",
+                            }
+                        }),
+                    )
+                } else {
+                    reject("Nothing given")
+                }
+            },
+        )
+    })
+}
+
+function syncGoogleSheets(
+    _category: Repository<CategoryEntity>,
+    _project: Repository<ProjectEntity>,
+    auth: OAuth2Client,
+    sheetID: string,
+) {
+    getSheet(auth, sheetID).then((sheet: Array<any>) => {
+        Promise.all(
+            sheet.map(async (row, i) => {
+                let category = await _category.findOne({ where: { name: row.category } })
+                if (category === void 0) {
+                    category = await _category.save({ name: row.category })
+                }
+
+                let project = await _project.findOne({ where: { name: row.name } })
+                delete row.category
+
+                try {
+                    if (project === void 0) {
+                        row.categoryId = category.id
+                        await _project.save(row)
+                    } else {
+                        delete row.externalLinks
```

Comment:
Allow links to be updated if they exist
```suggestion
row.externalLinks = project.externalLinks.map((link) => {
    return { ...link, url: row[link.name] }
})
```